grype/db/v6/name/java.go

```diff
@@ -0,0 +1,45 @@
+package name
+
+import (
+	"fmt"
+
+	grypePkg "github.com/anchore/grype/grype/pkg"
+	"github.com/anchore/grype/internal/log"
+	"github.com/anchore/grype/internal/stringutil"
+	"github.com/anchore/packageurl-go"
+)
+
+type JavaResolver struct {
```

Comment:
I would prefer this to somehow be in the Java matcher, by adding logic in different places like this it becomes harder to figure out what's actually going on during the matching process. It's my feeling that the matchers should be very much representative of the entire process of taking data from a package and mapping it to criteria used to search.

Reply:
yeah, this is probably going to be a grype v1 refactor, but this at least gets matching functional again for v6